### PR TITLE
tools: various venv fixes

### PR DESCRIPTION
**Pull request by mgfritch**
Fixes a few issues with handling of the python virtual environment.

### CODE REVIEW

**Comment by mgfritch** [comment]
Uses the venv when run under sudo:

```
$ sudo ./run_aquarium.sh 
Traceback (most recent call last):
  File "/usr/bin/uvicorn", line 11, in <module>
    load_entry_point('uvicorn==0.12.3', 'console_scripts', 'uvicorn')()
  File "/usr/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/uvicorn/main.py", line 357, in main
    run(**kwargs)
  File "/usr/lib/python3.8/site-packages/uvicorn/main.py", line 381, in run
    server.run()
  File "/usr/lib/python3.8/site-packages/uvicorn/_impl/asyncio.py", line 47, in run
    loop.run_until_complete(self.serve(sockets=sockets))
  File "/usr/lib64/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/usr/lib/python3.8/site-packages/uvicorn/_impl/asyncio.py", line 54, in serve
    config.load()
  File "/usr/lib/python3.8/site-packages/uvicorn/config.py", line 306, in load
    self.loaded_app = import_from_string(self.app)
  File "/usr/lib/python3.8/site-packages/uvicorn/importer.py", line 23, in import_from_string
    raise exc from None
  File "/usr/lib/python3.8/site-packages/uvicorn/importer.py", line 20, in import_from_string
    module = importlib.import_module(module_str)
  File "/usr/lib64/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "./aquarium.py", line 24, in <module>
    from fastapi import FastAPI
ModuleNotFoundError: No module named 'fastapi'


$ find venv -iname '*fastapi*'
venv/lib/python3.8/site-packages/fastapi
venv/lib/python3.8/site-packages/fastapi-0.63.0.dist-info
```

File: tools/setup-dev.sh [hunk 1]
```diff
@@ -103,20 +103,22 @@ git submodule update --init || exit 1
...
   ln -fs ../ceph.git/src/cephadm/cephadm src/gravel/cephadm/cephadm.bin
 
-if [[ ! -e "venv" ]]; then
-
-  # we need system site packages because librados python bindings appear to only
-  # be available as a package. It might be a good idea to compile it from the
-  # ceph repo we keep as a submodule, but it might be overkill at the moment?
-  python3 -m venv --system-site-packages venv || exit 1
+if [ -d venv ] ; then
+    >&2 echo "Detected an existing virtual environment - blowing it away!"
```
Comment (jhesketh) [inline tools/setup-dev.sh:107]: (nit) could prompt to do this

**Review by jhesketh — APPROVED** [review]

**Review by jecluis — APPROVED** [review]
I'm with @jhesketh that we could prompt to remove the venv, but this seems fine anyway. Let's see how annoying that is first.

**Comment by mgfritch** [comment]
added a quick and dirty yes/no prompt:

```
Detected an existing virtual environment:
  > /home/user/src/aquarium/venv
Blow it away?
```

**Review by jecluis — APPROVED** [review]

**Pull request merged**